controllers/admin/AdminImagesController.php

```diff
@@ -556,10 +629,24 @@ protected function _regenerateNewImages($dir, $type, $productsImages = false)
                                 $this->errors[] = $this->trans('Failed to resize image file (%filepath%)', ['%filepath%' => $dir . $image], 'Admin.Design.Notification');
                             }
 
+                            if ($generateAdditionalWebP) {
+                                ImageManager::resize($dir . $image, $newDir . substr(str_replace('_thumb.', '.', $image), 0, -4) . '-' . stripslashes($imageType['name']) . '.webp', (int) $imageType['width'], (int) $imageType['height'], 'webp', true);
```

Comment:
So many similar lines like this. You could think about wrapping this into a dedicated function.

Reply:
This class is already ugly. There is a PR to convert this controller to a symfony one and I suggested to merge this PR before implementing the new feature. I don't know if we have to take time to enhance this controller.